.github/github_workflow_scripts/purge_branch_protection_rules.py

```diff
@@ -0,0 +1,342 @@
+
+            table_body = tabulate.tabulate(
+                tabular_data=table_rows,
+                headers=headers,
+                tablefmt='github'
+            )
+            markdown_content = f"{header}\n\n{table_body}\n"
+
+        logger.debug(f"Writing deleted jobs summary to Markdown to file '{fp}'...")
+        logger.debug(markdown_content)
+        fp.write_text(markdown_content)
+        logger.debug("Finished writing jobs summary to Markdown to file")
+    else:
+        logger.info(
+            f"Environmental variable '{GH_JOB_SUMMARY_ENV_VAR}' not set. Skipping writing job summary for deleted rules...")
+
+
+def get_token():
+    """
+    Helper method to retrieve the GitHub token
+    from the environmental variables.
+
+    Returns:
+    - `token` (``str``): The GitHub token.
+
+    Raises:
+    `OSError` if the `GITHUB_TOKEN` env var is not set.
+    """
+    token = os.getenv(GH_TOKEN_ENV_VAR)
+    if not token:
+        raise OSError(f"Error: The '{GH_TOKEN_ENV_VAR}' environment variable is not set.")
+    return token
+
+
+def send_request(gh_requester: github.Requester.Requester, query: str, variables: dict[str, str]) -> dict[str, str]:
+    """
+    Wrapper function to send a request to the GraphQL endpoint.
+
+    Arguments:
+    - `query` (``str``): The query to send.
+    - `variables` (``dict[str, str]``): The variables to send the query.
+
+    Returns:
+    - A `dict[str, str]` with the response.
+
+    Raises:
+    If the request fails, we raise
+    """
+
+    logger.debug(f"Sending GraphQL request...\n{query=}\n{variables=}\n")
+
+    response_headers, response_data = gh_requester.graphql_query(  # type:ignore[attr-defined]
+        query=query,
+        variables=variables
+    )
+
+    logger.debug(f"Response received:\n{response_data=}\n{response_headers=}")
+
+    return response_data
+
+
+# API Functions
+def purge_branch_protection_rules(
+    gh_requester: github.Requester.Requester,
+    rules: list[BranchProtectionRule]
+) -> list[BranchProtectionRule]:
+    """
+    Delete all branch protection rules except for the ones
+    specified in the exception list.
+
+    Arguments:
+    - `gh_requester` (``github.Requester.Requester``): The instance
+    of the GitHub client.
+    - `rules` (``list[BranchProtectionRule]``): The rules to iterate over
+    and delete.
+
+    Returns:
+    - `list[BranchProtectionRule]` that were deleted
+    """
+
+    deleted: list[BranchProtectionRule] = []
+
+    for i, rule in enumerate(rules):
+        if should_delete_rule(rule):
+            logger.info(f"Deleting {i}/{len(rules)}: {rule}...")
+
+            query = DELETE_BRANCH_PROTECTION_RULE_QUERY_TEMPLATE
+            variables = {
+                "branchProtectionRuleId": rule.id
+            }
+            send_request(gh_requester, query, variables)
+            logger.info(f"{rule} was deleted successfully.")
+            deleted.append(rule)
```

Comment:
Try except?

Reply:
Sounds good, in all other cases let's keep the exceptions aside and print them in the end, we can also fail the flow after everything was done.

Reply:
https://github.com/demisto/content/pull/35604/commits/12bc53f7c6cb6b4f70d2c6d2e1d56bda8e1f37b6

Reply:
If any deletions fail, we will print out how many did. In addition, the summary will show if the rule was deleted and if there's an error:

<img width="1360" alt="image" src="https://github.com/user-attachments/assets/d7c9ca70-90b0-481c-b0c7-b9c627217130">